refactor: controller 공통 Response 적용

mixedsider opened the pull request:
## 📌 PR 요약
- Controller Response 적용
- Book API 권한 추가
- ForbiddenRequestException 추가

## 🔗 관련 이슈
- 관련된 이슈 번호를 연결해주세요.
    - closed #46 

## 🛠️ 변경 사항
- 주요 변경 사항을 bullet으로 작성해주세요.
    - Book API 권한 추가
    - AuthUser 단일 권한 확인용 메소드 (getRole) 추가
    - ForbiddenRequestException 추가

## 📸 스크린샷 (선택)
UI 변경 사항이 있다면 스크린샷을 첨부해주세요.

| 기능 | 스크린샷 |
| --- | --- |
| PNG | ![스크린샷](사진을 넣어주세요) |

## ⚠️ 주의 사항 (선택)
리뷰어가 알아야 할 중요한 사항이나 주의할 점을 작성해주세요.

## ✅ 체크리스트
PR 작성자가 확인해야 할 항목입니다:
- [ ] 코드가 정상적으로 동작하는지 확인했습니다.
- [ ] 관련 문서를 업데이트했습니다.
- [x] 코드 리뷰어를 등록했습니다.
- [x] Labels를 등록했습니다.

mixedsider committed:
refactor: controller 공통 Response 적용

dereck-jun commented on src/main/java/xyz/tomorrowlearncamp/bookking/domain/order/controller/OrderController.java:
```diff
@@ -16,13 +19,12 @@ public class OrderController {
-    // todo : 공통 response 사용
-    @GetMapping("/v1/orders/myinfo")
-    public ResponseEntity<Page<OrderResponse>> getMyOrders(
+    @GetMapping("/v1/orders/myInfo")
```
`myInfo` -> `my-info`

mixedsider replied:
check

dereck-jun commented on src/main/java/xyz/tomorrowlearncamp/bookking/domain/review/controller/ReviewController.java:
```diff
@@ -20,50 +20,50 @@ public class ReviewController {
     @PatchMapping("/v1/books/{bookId}/reviews/{reviewId}")
-    public ResponseEntity<String> updateReview(
+    public Response<String> updateReview(
             @PathVariable Long bookId,
             @PathVariable Long reviewId,
             @Valid @RequestBody ReviewUpdateRequest request,
             @AuthenticationPrincipal AuthUser authUser
     ) {
         reviewService.updateReview(authUser.getUserId(), bookId, reviewId, request);
-        return ResponseEntity.ok("리뷰가 성공적으로 수정되었습니다.");
+        return Response.success("리뷰가 성공적으로 수정되었습니다.");
```
다른 도메인의 수정 API와 마찬가지로 `void`로 통일하는 것이 더 좋아보입니다.

dereck-jun commented on src/main/java/xyz/tomorrowlearncamp/bookking/domain/review/controller/ReviewController.java:
```diff
@@ -20,50 +20,50 @@ public class ReviewController {
     @DeleteMapping("/v1/books/{bookId}/reviews/{reviewId}")
-    public ResponseEntity<String> deleteReview(
+    public Response<String> deleteReview(
             @PathVariable Long bookId,
             @PathVariable Long reviewId,
             @AuthenticationPrincipal AuthUser authUser
     ) {
         reviewService.deleteReview(authUser.getUserId(), bookId, reviewId);
-        return ResponseEntity.ok("리뷰가 성공적으로 삭제되었습니다.");
+        return Response.success("리뷰가 성공적으로 삭제되었습니다.");
```
위와 동일

dereck-jun commented on src/main/java/xyz/tomorrowlearncamp/bookking/domain/review/controller/ReviewController.java:
```diff
@@ -20,50 +20,50 @@ public class ReviewController {
+        return Response.success("리뷰가 성공적으로 삭제되었습니다.");
     }
 
     @GetMapping("/v1/reviews/myinfo")
```
`myinfo` -> `my-info`

dereck-jun commented on src/main/java/xyz/tomorrowlearncamp/bookking/domain/user/auth/controller/AuthController.java:
```diff
@@ -28,18 +30,18 @@ public class AuthController {
     private final JwtProvider jwtProvider;
 
     @PostMapping("/v1/auth/signup")
```
개인적으로 register-login 또는 signup-signin 같이 비슷한 느낌으로 맞추는 것이 좋을 것 같습니다.

dereck-jun commented on src/main/java/xyz/tomorrowlearncamp/bookking/domain/user/auth/controller/AuthController.java:
```diff
@@ -28,18 +30,18 @@ public class AuthController {
+        return Response.success(authService.signup(request));
     }
 
     @PostMapping("/v1/auth/login")
```
위와 동일

dereck-jun commented on src/main/java/xyz/tomorrowlearncamp/bookking/domain/user/controller/UserController.java:
```diff
@@ -3,41 +3,40 @@
 public class UserController {
     private final UserService userService;
 
     @GetMapping("/v1/users/myInfo")
```
`myInfo` -> `my-info`

dereck-jun commented on src/main/java/xyz/tomorrowlearncamp/bookking/domain/user/controller/UserController.java:
```diff
@@ -3,41 +3,40 @@
+        return Response.success(response);
     }
 
     @PatchMapping("/v1/users/myInfo")
```
위와 동일

dereck-jun commented on src/main/java/xyz/tomorrowlearncamp/bookking/domain/user/controller/UserController.java:
```diff
@@ -3,41 +3,40 @@
+        return Response.success(updatingUser);
     }
 
     @PatchMapping("/v1/users/{userId}/role")
```
accessToken으로 본인을 인증하기 때문에 userId PathVariable은 없어도 될 것 같습니다.

dereck-jun commented on src/main/java/xyz/tomorrowlearncamp/bookking/domain/user/controller/UserController.java:
```diff
@@ -47,25 +46,25 @@ public ResponseEntity<UserResponse> updateUserRole(
+        return Response.success(updateUser);
     }
 
     @DeleteMapping("/v1/users/{userId}")
```
위와 동일

dereck-jun reviewed (approved):
공통 Response 적용인 것은 알지만 어떤 것은 API endpoint가 변경되어 있는 것도 있어서 우선 리뷰 남겨두었습니다.

mixedsider committed:
Merge branch 'develop' into refactor/controller

# Conflicts:
#	src/main/java/xyz/tomorrowlearncamp/bookking/domain/common/enums/ErrorMessage.java

mixedsider committed:
fix: Controller 및 text 코드 수정

- returnType String -> Void
- returnType Void이면 @ResponseStatus(HttpStatus.OK) 추

dereck-jun commented on src/main/java/xyz/tomorrowlearncamp/bookking/domain/review/controller/ReviewController.java:
```diff
@@ -20,50 +21,50 @@ public class ReviewController {
 
     @PatchMapping("/v1/books/{bookId}/reviews/{reviewId}")
-    public ResponseEntity<String> updateReview(
+    @ResponseStatus(HttpStatus.OK)
```
void만 정의하는 경우 httpstatus 204가 더 적합하다고 생각합니다.

mixedsider committed:
fix: ERD 변경으로 인한 entity 수정

sjMun09 reviewed (approved):
LGTM !

mixedsider committed:
fix: order entity 원복
